Framework/Algorithms/src/WorkspaceJoiners.cpp

```diff
@@ -172,13 +172,16 @@ MatrixWorkspace_sptr WorkspaceJoiners::execEvent() {
  */
 void WorkspaceJoiners::validateInputs(const MatrixWorkspace &ws1,
                                       const MatrixWorkspace &ws2) {
-  // This is the full check for common binning
-  if (!WorkspaceHelpers::commonBoundaries(ws1) ||
-      !WorkspaceHelpers::commonBoundaries(ws2)) {
-    g_log.error(
-        "Both input workspaces must have common binning for all their spectra");
-    throw std::invalid_argument(
-        "Both input workspaces must have common binning for all their spectra");
+  // Workspaces with point data are allowed to have different binning
```

Comment:
I'm not this is exactly what we want here. Point data and event workspaces are different things. `EventWorkspace` is a special type of `MatrixWorkspace` that stores collections of signal/timestamp pairs per spectrum. Histograms of these collections are then generated on the fly from a current set of bin boundaries. Here the check is unimportant because of this on-the-fly generation.

For the `MatrixWorkspace` case, regarless of point or histogram status, the algorithm appends the spectra from the rhs to the lfs workspace but uses the X values exclusively from the lhs. If they are not the same in both cases then the numbers at the end could be misleading.

Perhaps the easiest way to solve this is to add a flag to `validateInputs` to disable the bin checking and then set this to `true` on line 58 of `ConjoinWorkspaces.cpp` as we already have checked for event vs non-event here?